test/topology_experimental_raft/test_crash_coordinator_before_streaming.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+import asyncio
+import logging
+import re
+import time
+from test.pylib.internal_types import ServerInfo
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.scylla_cluster import Cluster, ReplaceConfig
+from test.pylib.util import wait_for_cql_and_get_hosts
+from cassandra.cluster import SimpleStatement, ConsistencyLevel
+from test.topology.util import check_token_ring_and_group0_consistency
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_kill_coordinator_during_op(manager: ManagerClient) -> None:
+    """ Kill coordinator with error injection while topology operation is running for cluster: decommission,
+    bootstrap, removenode, replace.
+
+    1. Find the coordinator node.
+    2. Inject an error to abort coordinator before streaming starts
+    3. Start the operation on target node
+    4. Wait for the operation to abort
+    5. Check if the new coordinator has been elected
+    6. Start the old coordinator node
+   
+    Topology operation is expected to fail and cluster is rolled back
+    to previous state.
+
+    | Operation | Coverage |
+    | Decommission | done |
+    | Removenode | done |
+    | Bootstrap | #14804 |
+    | Replace | #14804 |
+
+    2 operations should be covered after issue will be fixed
+
+    """
+    # Decrease the failure detector threshold so we don't have to wait for too long.
+    config = {
+        'failure_detector_timeout_in_ms': 2000
+    }
+    nodes = [await manager.server_add(config=config) for _ in range(3)]
+    cql = manager.get_cql()
+
+    coordinators_ids = await get_coordinator_host_ids(manager)
+    assert len(coordinators_ids) == 1, "At least 1 coordinator id should be found"
+
+    # kill coordinator during decommission
+    coordinator_host = await get_coordinator_host(coordinators_ids[0], manager)
+    other_nodes = [srv for srv in nodes if srv.server_id != coordinator_host.server_id]
+
+    await manager.api.enable_injection(coordinator_host.ip_addr, "crash_coordinator_before_stream", one_shot=True)
+    await manager.decommission_node(server_id=other_nodes[-1].server_id, expected_error="Decommission failed. See earlier errors")
+
+    coordinators_ids = await get_coordinator_host_ids(manager)
+    assert len(coordinators_ids) == 2, "No new record about the coordinator's election"
```

Comment:
There is no guarantee how fast the new coordinator will be elected. It should generally take less than 2 seconds, but we cannot depend on such timing.

Now the test will race with Scylla and sometimes you will reach this check before Scylla manages to elect the new coordinator.

I suggest using `wait_for` with reasonable timeout (60 seconds?) to wait for the condition that a new coordinator was elected.